### PR TITLE
Using Chromium's spellchecker

### LINKED ISSUE #9880: Use Chromium spell checker
## Test plan
See https://github.com/brave/browser-laptop/pull/9951

## Original issue description
deprecate electron spell checker

### PR DESCRIPTION
**Pull request by darkdh**
Test Plan:
--
a. Suggestions

1. Go to `about:styles` 
2. Type `braave` in textbox
3. There should be redline under `braave`
4. Open context menu on `braave`
5. You can see `bravo`, `brave`, `Brava`
6. Click on `brave`
7. The text should be replaced with `brave` with no redline

b. Learn Spelling
1. Go to `about:styles` 
2. Type `braave` in textbox
3. There should be redline under `braave`
4. Open context menu on `braave`
5. Click `Learn Spelling`
6. `braave` will no longer be redlined 
7.  Next time you type `braave`, you will not see redline

c. Forget Learned Spelling
1. Go to `about:styles` 
2. Type `braave` in textbox
3. There should be redline under `braave`
4. Open context menu on `braave`
5. Click `Learn Spelling`
6. `braave` will no longer be redlined 
7. Open context menu on `braave`
8. Click `Forget Learned Spelling`
9.  Next time you type `braave`, you will see redline under it

d. Legacy data migration
1. Use older Brave to `Learn Spelling` for `braave` and `Ignore Spelling` for `braaave`
2. Update to the version with chromium spellchecker
3. There will be no `dictionary` structure in session store
4. You can see those two words will be added to `Custom Dictionary.txt` under user folder
5. And try to type `braave` and `braaave` in `about:styles`
6. They should be valid words
---
Auditors: @bridiver, @bbondy, @bsclifton 

fixes #9880 

requires https://github.com/brave/muon/pull/244

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Reviewer Checklist:

Tests

- [x] Adequate test coverage exists to prevent regressions
- [x] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [x] New files have MPL2 license header

### CODE REVIEW

**Comment by darkdh** [comment]
working on migrating old custom dictionary data

File: app/browser/menu.js [hunk 1]
```diff
@@ -266,6 +266,7 @@ const createViewSubmenu = () => {
       label: locale.translation('toggleDeveloperTools'),
       accelerator: isDarwin ? 'Cmd+Alt+I' : 'Ctrl+Shift+I',
       click: function () {
+        const appStore = require('../../js/stores/appStore')
```
Comment (NejcZdovc) [inline app/browser/menu.js]: This was already fixed, so if you rebase to the master, you can remove this code :wink:

Comment (darkdh) [inline app/browser/menu.js]: ah, I forgot to remove the workaround. thanks

File: js/constants/messages.js [hunk 2]
```diff
@@ -77,7 +77,6 @@ const messages = {
   RELOAD: _,
   DETACH: _,
   IS_MISSPELLED: _, /** @arg {string} word, the word to check */
```
Comment (bridiver) [inline js/constants/messages.js]: this should also be removed

File: app/spellCheck.js [hunk 3]
```diff
@@ -35,13 +35,6 @@ module.exports.init = () => {
     }
```
Comment (bridiver) [inline app/spellCheck.js]: doesn't this whole file go away?

File: js/contextMenus.js [hunk 4]
```diff
@@ -543,17 +543,19 @@ function getMisspelledSuggestions (selection, isMisspelled, suggestions) {
     if (isMisspelled) {
       template.push({
         label: locale.translation('learnSpelling'),
-        click: () => {
-          appActions.addWord(selection, true)
-          // This is needed so the underline goes away
-          webviewActions.replace(selection)
+        click: (item, focusedWindow) => {
+          if (focusedWindow) {
+            focusedWindow.webContents.addWord(selection)
```
Comment (bridiver) [inline js/contextMenus.js]: `focusedWindow` should be deprecated and context menu items should use actions by default. I would use something like `appActions.learnSpelling` and `appActions.forgetLearnedSpelling` and pass the tabId (always set in muon `HandleContextMenu` and retrieved from nodeProps.properties)

File: app/spellCheck.js (hunk 3 above)
Comment (darkdh) [inline app/spellCheck.js]: :sweat_smile: yeah, I forgot to remove

File: app/browser/reducers/spellCheckerReducer.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+'use strict'
+
+const Immutable = require('immutable')
+const appConstants = require('../../../js/constants/appConstants')
+const {makeImmutable} = require('../../common/state/immutableUtil')
+const {getWebContents} = require('../webContentsCache')
+const spellChecker = require('../../spellChecker')
+
+const migrate = (state) => {
```
Comment (bridiver) [inline app/browser/reducers/spellCheckerReducer.js]: the other data migrations are in session store. We don't want it to get too bloated, but if we don't remove `dictionary` from the saved state this migration will run on every restart

**Comment by bsclifton** [comment]
Will check this out once Muon 4.4.0 prebuilt is ready to go 😄

File: app/spellCheck.js [hunk 6]
```diff
@@ -1,90 +0,0 @@
...
-      lastAddedWords = appStore.getState().getIn(['dictionary', 'addedWords'])
-    }
-  })
-
-  // Thank you Slack team.
-  // NB: This is to work around tinyspeck/slack-winssb#267, where contractions
-  // are incorrectly marked as spelling errors. This lets people get away with
-  // incorrectly spelled contracted words, but it's the best we can do for now.
-  const contractions = [
-    "ain't", "aren't", "can't", "could've", "couldn't", "couldn't've", "didn't", "doesn't", "don't", "hadn't",
```
Comment (bsclifton) [inline app/spellCheck.js]: pfft new spell checker ain't got these words, y'all

**Comment by Jacalz** [comment]
This fixed #7967 thank you guys so so much :+1:

**Comment by luixxiul** [comment]
@darkdh does this also close #2183 and #4697?

**Pull request merged**